packages/ember-data/lib/adapters/rest-adapter.js

```diff
@@ -598,7 +598,12 @@ export default Adapter.extend(BuildURLMixin, {
   },
 
   // http://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers
-  maxUrlLength: 2048,
+  maxURLLength: 2048,
+
+  maxUrlLength: Ember.computed(function() {
```

Comment:
Shouldn't turn it into a computed if even we weren't calling get on it. You can deprecate using deprecateProperty/making a prop getter similar to https://github.com/emberjs/data/blob/master/packages/ember-data/lib/system/snapshot.js#L39